src/Drivers/CMakeLists.txt

```
@@ -39,5 +39,5 @@ endif()
 
 if(HIOP_BUILD_SHARED AND NOT HIOP_USE_GPU)
   add_executable(nlpMDS_cex4.exe nlpMDS_ex4.c)
-  target_link_libraries(nlpMDS_cex4.exe hiop_shared)
+  target_link_libraries(nlpMDS_cex4.exe HiOp::SHARED)
```

```suggestion
  target_link_libraries(nlpMDS_cex4.exe HiOp::HiOp)
```
I am not sure if having `SHARED` target is a good idea. I would think that `HiOp::HiOp` target should install shared, static or both libraries depending on configuration option.

---

IIUC the C interface needs the shared library which is why this test is only built `if(HIOP_BUILD_SHARED)` - @cnpetra is this correct?. `HiOp::HiOp` will point to shared by default **_and to static only if shared is not being built_** so that the alias target always points to a valid library. This is why I don't just link against `HiOp::HiOp`.